datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/load/AspectResolver.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.datahub.graphql.resolvers.load;
+
+import com.linkedin.datahub.graphql.AspectLoadKey;
+import com.linkedin.datahub.graphql.generated.Aspect;
+import com.linkedin.datahub.graphql.generated.Entity;
+import com.linkedin.datahub.graphql.resolvers.ResolverUtils;
+import com.linkedin.datahub.graphql.types.aspect.AspectMapper;
+import com.linkedin.metadata.aspect.AspectWithMetadata;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.concurrent.CompletableFuture;
+import org.dataloader.DataLoader;
+
+
+/**
+ * Generic GraphQL resolver responsible for
+ *
+ *    1. Generating a single input AspectLoadKey.
+ *    2. Resolving a single {@link Aspect}.
+ *
+ */
+public class AspectResolver implements DataFetcher<CompletableFuture<Aspect>> {
+
+    public AspectResolver() {
+    }
+
+    @Override
+    public CompletableFuture<Aspect> get(DataFetchingEnvironment environment) {
+        final DataLoader<AspectLoadKey, Aspect> loader = environment.getDataLoaderRegistry().getDataLoader("Aspect");
+
+        String fieldName = environment.getField().getName();
+        Long version = environment.getArgument("version");
+        String urn = ((Entity) environment.getSource()).getUrn();
```

Comment:
Will this _always_ be invoked in the context of a broader entity fetch? I guess it must? 

We could also have a dedicated "getVersionedAspect" method that requires version, urn, etc to do this

Reply:
`getVersionedAspect` is something we could consider- although the type safety of that will be a bit off- the return type will need to be `Aspect` and we will always need to type check it on the client.

Reply:
Yes - similar to our entity resource